stko/calculators/torsion_calculators.py

```diff
@@ -114,15 +115,64 @@ class ConstructedMoleculeTorsionCalculator(TorsionCalculator):
             )
 
     """
+    
+    def calculate(self, mol):
+        """extract torsions with rdkit, then match to building blocks
+        """
+        def get_atom_maps():
+            """
+            map from building block atom ids to constructed molecule atoms for a
+            specified building block id
+            """
+            atom_maps = defaultdict(dict)
+            for atom_info in mol.get_atom_infos():
+                current_atom_map = atom_maps[atom_info.get_building_block_id()]
+                current_atom_map[atom_info.get_building_block_atom().get_id()] = atom_info.get_atom()
+            return atom_maps
+        
+        torsions = list(next(super().calculate(mol)))
+        atom_maps = get_atom_maps()
+        
+        # loop over torsions, updating each to match a building block
+        # torsion if possible
+        for i, torsion in enumerate(torsions):
+            # atom ids, atom infos, and building block ids of current
+            # torsion in constructed molecule
+            atom_ids = list(torsion.get_atom_ids())
+            atom_infos = list(mol.get_atom_infos(atom_ids))
+            build_block_ids = [atom_info.get_building_block_id() for atom_info in atom_infos]
+            
+            # check if two central atoms of torsion are from the same building block
+            # if not, leave this torsion alone
+            if build_block_ids[1] is None:
+                continue
+            if atom_infos[1].get_building_block_id() != atom_infos[2].get_building_block_id():
```

Comment:
I see what you mean - I think this could be problematic for edge cases (just imagining a weird reaction that someone is unlikely to actually do). In that I would expect that any torsions that include a constructed bond would be removed... but then they are equivalent to the bond that may be deleted. 

Sorry for the ramblings. I see your point, just needs to be made clear in the docs that the filtering is based on the central bond only.